src/llama_stack_client/lib/direct/direct.py

```diff
@@ -17,15 +20,21 @@
 
 
 class LlamaStackDirectClient(LlamaStackClient):
-    def __init__(self, config: StackRunConfig, **kwargs):
+    def __init__(self, config: StackRunConfig | str, **kwargs):
         super().__init__(**kwargs)
         self.endpoints = get_all_api_endpoints()
-        self.config = config
+
+        # Allow initialization with template name string
+        if isinstance(config, str):
+            self.config = get_stack_run_config_from_template(config)
+        else:
+            self.config = config
+
         self.dist_registry = None
         self.impls = None
 
     async def initialize(self) -> None:
-        self.dist_registry, _ = await create_dist_registry(self.config)
+        self.dist_registry, _ = await create_dist_registry(self.config.metadata_store, image_name="")
```

Comment:
I think we should nuke all of this and move it all to use `construct_stack()` like in the server code. Where is replace_env_vars() happening if the user provides just a yaml btw? It should still happen!

Reply:
done.